FabLabKasse/gui.py

```diff
@@ -817,8 +835,50 @@ def leaveSearch(self, keepResultsVisible=False):
         # Give focus to lineEdit
         self.lineEdit.setFocus()
 
+    def _check_idle(self):
+        """checks whether the GUI is idle for a great time span
+
+        Uses the information from screensaver to check whether the GUI is idle for a hardcoded time span.
+        If the GUI is considered idle, then true is returned.
+        :rtype: bool
+        :return: true if GUI is idle
+        """
+        idle_state = self.idleTracker.check_idle()
+        # check_idle() returns a tupel (state_change, suggested_time_till_next_check, idle_time)
+        # the state "idle" is entered after the time configured in self.CATEGORY_VIEW_RESET_TIME
+        idle_keyword = "idle"
+        if idle_state[0] == idle_keyword:
+            return True
+        elif idle_state[0] is None and self.idleTracker.last_state == idle_keyword:
+            return True
+        else:
+            return False
 
+    def _reset_if_idle(self):
+        """resets the category-view of the GUI if it is idle for a certain timespan"""
```

Comment:
+doc:
This function might be called even if another dialog like payup is currently active. Therefore the current order is not modified or updated, as result might be undefined during payup.